pkg/frontend/openshiftcluster_preflightvalidation_test.go

```diff
@@ -6,96 +6,342 @@ package frontend
 )
 
 func TestPreflightValidation(t *testing.T) {
 	ctx := context.Background()
 	mockSubID := "00000000-0000-0000-0000-000000000000"
+	apiVersion := "2020-04-30"
+	clusterId := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName"
+	location := "eastus"
+	defaultProfile := "default"
+	resourceGroup := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroupTest"
+	netProfile := "10.128.0.0/14"
+	encryptionSet := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/ms-eastus/providers/Microsoft.Compute/diskEncryptionSets/ms-eastus-disk-encryption-set"
+	masterSub := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/vnetResourceGroup/providers/Microsoft.Network/virtualNetworks/vnet/subnets/master"
+	workerSub := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/vnetResourceGroup/providers/Microsoft.Network/virtualNetworks/vnet/subnets/worker"
+
+	preflightPayload := []byte(fmt.Sprintf(`
+	{
+		"apiVersion": "%s",
+		"id": "%s",
+		"name": "%s",
+		"type": "%s",
+		"location": "%s",
+		"properties": {
+			"clusterProfile": {
+				"domain": "%s",
+				"resourceGroupId": "%s",
+				"fipsValidatedModules": "%s",
+				"version": "%s"
+			},
+			"consoleProfile": {},
+			"servicePrincipalProfile": {
+				"clientId": "%s",
+				"clientSecret": "%s"
+			},
+			"networkProfile": {
+				"podCidr": "%s",
+				"serviceCidr": "%s"
+			},
+			"masterProfile": {
+				"vmSize": "%s",
+				"subnetId": "%s",
+				"encryptionAtHost": "%s",
+				"diskEncryptionSetId": "%s"
+			},
+			"workerProfiles": [
+			{
+				"name": "%s",
+				"vmSize": "%s",
+				"diskSizeGB": %v,
+				"encryptionAtHost": "%s",
+				"subnetId": "%s",
+				"count": %v,
+				"diskEncryptionSetId": "%s"
+			}
+			],
+			"apiserverProfile": {
+				"visibility": "%s"
+			},
+			"ingressProfiles": [
+			{
+				"name": "%s",
+				"visibility": "%s",
+				"IP": "%s"
+			}
+			]
+		}
+	}
+	`, apiVersion, clusterId, api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Name,
+		api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Type,
+		location, defaultProfile, resourceGroup, api.EncryptionAtHostEnabled, version.DefaultInstallStream.Version.String(),
+		mockSubID, mockSubID, netProfile, netProfile, api.VMSizeStandardD32sV3, masterSub,
+		api.EncryptionAtHostEnabled, encryptionSet,
+		api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Properties.WorkerProfiles[0].Name, api.VMSizeStandardD32sV3,
+		api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Properties.WorkerProfiles[0].DiskSizeGB,
+		api.EncryptionAtHostEnabled, workerSub,
+		api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Properties.WorkerProfiles[0].Count,
+		encryptionSet, api.VisibilityPublic, defaultProfile, api.VisibilityPublic,
+		api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Properties.IngressProfiles[0].IP))
```

Comment:
nit: 28 ordered parameters is going to be hard to maintain over time - if it's easy enough, can we convert this to a [golang string template](https://pkg.go.dev/text/template#pkg-overview), and then populate it with named parameters rather than ordered ones?